### PR TITLE
add auto-update for imgareaselect

### LINKED ISSUE #6207: [Request] Add other versions of jquery.imgareaselect
jquery.imgareaselect
https://cdnjs.com/libraries/imgareaselect
https://github.com/odyniec/imgareaselect

### PR DESCRIPTION
**Pull request by x09326**
PR for #6207 
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/odyniec/imgareaselect
 * Official website (optional, not the repository): http://odyniec.net/projects/imgareaselect/
 * NPM package url (optional):
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 567
   - Count of watchers: 40
   - Count of forks: 166
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #6207

### CODE REVIEW

**Comment by x09326** [comment]
@dannyxx001 please help me review this enrichment, thank you.
The auto-update is following the lib [repo](https://github.com/odyniec/imgareaselect), you can go there to check.

File: ajax/libs/imgareaselect/package.json [hunk 1]
```diff
@@ -26,5 +26,13 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/odyniec/imgareaselect"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/odyniec/imgareaselect.git",
+    "basePath": "distfiles",
+    "files": [
+      "**!(*.txt)/*"
```
Comment (PeterDaveHello) [inline ajax/libs/imgareaselect/package.json]: Should be `"**/*.!(txt)/"`

Comment (x09326) [inline ajax/libs/imgareaselect/package.json]: But I used git auto-update to test it, then I didn't get the ```.txt``` files.
The ```.txt``` files are in the ```distfiles``` directory, not in the ```css``` and ```scripts``` directory.

Comment (PeterDaveHello) [inline ajax/libs/imgareaselect/package.json]: Didn't get your problem, did you mean `"**/*.!(txt)"` doesn't work?

Comment (x09326) [inline ajax/libs/imgareaselect/package.json]: Yes, if use  ```"**/*.!(txt)"``` then we will get the ```txt``` files but we don't need them.

Comment (PeterDaveHello) [inline ajax/libs/imgareaselect/package.json]: Okay, I'll try it later.

Comment (PeterDaveHello) [inline ajax/libs/imgareaselect/package.json]: I tried with `"**/*.!(txt)"` but didn't see any txt files been included, please confirm that.

Comment (PeterDaveHello) [inline ajax/libs/imgareaselect/package.json]: `jquery.min.js` should also been excluded, it's such a well known library been hosted everywhere, and also on cdnjs. We don't need to host it here.

Comment (x09326) [inline ajax/libs/imgareaselect/package.json]: So, can I change the files to ```"css/*``` ?

Comment (PeterDaveHello) [inline ajax/libs/imgareaselect/package.json]: `"**/*.!(txt)"` will be better, you don't know when will the author add other files in `img/` or `js/`, I'll like to use rule with more flexible.

Comment (x09326) [inline ajax/libs/imgareaselect/package.json]: How about ```"**!(scripts)/*.!(txt)"``` ? Because you said that we don't need the ```jquery.min.js```.

Comment (PeterDaveHello) [inline ajax/libs/imgareaselect/package.json]: Oops, forgot jquery, what about this:
`"**/!(*.txt|jquery.min.js)"`

**Comment by x09326** [comment]
@PeterDaveHello 
I have already fixed those problems. Please help me review it, thanks.

**Pull request merged**